bika/lims/browser/analysisrequest/publish.py

```diff
@@ -1391,6 +1412,10 @@ def _analysis_data(self, analysis, decimalmark=None):
         # is out of range. The second value (dismissed here) is a bool that
         # indicates if the result is out of shoulders
         andict['outofrange'] = is_out_of_range(analysis)[0]
+        andict['interims'] = hasattr(analysis, 'getInterimFields') \
+                   and analysis.getInterimFields() or []
+        if len(andict['interims']) > 0:
+            andict['interims'] = [a for a in andict['interims'] if 'report' in a]
```

Comment:
I think you need to check here if the interim's "report" setting is True, otherwise it will always display all interims. Something similar to could work:
```python
interims = analysis.getInterimFields()
andict['interims'] = filter(lambda interim: interim.get('report', False), interims)
```
Note that there is no need to use `hasattr`, cause [`InterimFields` is defined in `AbstractAnalysis`](https://github.com/senaite/senaite.core/blob/master/bika/lims/content/abstractanalysis.py#L121-L135), so is always available, regardless of the analysis type (`DuplicateAnalysis`, `ReferenceAnalysis`, `Analysis`, etc.)